### PR TITLE
add db reconnect process on __init__.py

### PR DESCRIPTION
**Pull request by crazychengmm**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [ ] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

If mysql was restarted,it need to restart librenms.service ,service could't reconnect db.

### CODE REVIEW

**Comment by crazychengmm** [comment]
If mysql was restarted,it need to restart librenms.service ,service could't reconnect db.

File: LibreNMS/__init__.py [hunk 1]
```diff
@@ -106,14 +130,11 @@ def query(self, query, args=None):
...
-            cursor.close()
-            return cursor
+            return self.__query__(query, args)
         except Exception as e:
-            critical("DB Connection exception {}".format(e))
-            self.close()
-            raise
+            critical("ERROR: Mysql databse has problem !{}".format(e))
+            times.sleep(20)
+            return self.__query__(query, args)
```
Comment (murrant) [inline LibreNMS/__init__.py]: What happens after it is not back after it the sleep(20)?  Looks like it fails the same way.

**Comment by murrant** [comment]
Previous code accepted that it was ok for mysql to fail and it should try to connect each query.  Sleeping for 20(s?) could be pretty bad in some places.

**Comment by crazychengmm** [comment]
> Previous code accepted that it was ok for mysql to fail and it should try to connect each query. Sleeping for 20(s?) could be pretty bad in some places.

If mysql have been restarted, librenms need to restart too.Service could't reconnect the db in previous code. 20S may be too long ,it could be 2s or other.

**Comment by murrant** [comment]
Here is a log showing polling running before, during, and after a mysql stop.  It seems to be working.
```
Poller_0-52(INFO):Polling device 1
Poller_0-52(INFO):Completed poller run for 1 in 6.63s
Alerting(INFO):Checking alerts
Billing(INFO):Calculating billing
Alerting(INFO):Completed alerting run for alerts in 0.33s
Billing(INFO):Completed billing run for calculate in 0.33s
Alerting(INFO):Checking alerts
Billing(INFO):Calculating billing
Alerting(INFO):Completed alerting run for alerts in 0.36s
Billing(INFO):Completed billing run for calculate in 0.32s
Alerting(INFO):Checking alerts
Billing(INFO):Calculating billing
Alerting(INFO):Completed alerting run for alerts in 0.42s
Billing(INFO):Completed billing run for calculate in 0.35s
Alerting(INFO):Checking alerts
Billing(INFO):Calculating billing
Alerting(INFO):Completed alerting run for alerts in 0.35s
Billing(INFO):Completed billing run for calculate in 0.31s
host(CRITICAL):DB Connection exception (2013, 'Lost connection to MySQL server during query')
host(INFO):Using pure python SQL client
host(CRITICAL):ERROR: Could not connect to MySQL database! (2003, "Can't connect to MySQL server on 'localhost' ([Errno 111] Connection refused)")
host(CRITICAL):DB Connection exception (2003, "Can't connect to MySQL server on 'localhost' ([Errno 111] Connection refused)")
host(INFO):Using pure python SQL client
host(CRITICAL):ERROR: Could not connect to MySQL database! (2003, "Can't connect to MySQL server on 'localhost' ([Errno 111] Connection refused)")
host(CRITICAL):DB Connection exception (2003, "Can't connect to MySQL server on 'localhost' ([Errno 111] Connection refused)")
host(INFO):Using pure python SQL client
host(CRITICAL):ERROR: Could not connect to MySQL database! (2003, "Can't connect to MySQL server on 'localhost' ([Errno 111] Connection refused)")
host(CRITICAL):DB Connection exception (2003, "Can't connect to MySQL server on 'localhost' ([Errno 111] Connection refused)")
host(INFO):Using pure python SQL client
host(CRITICAL):ERROR: Could not connect to MySQL database! (2003, "Can't connect to MySQL server on 'localhost' ([Errno 111] Connection refused)")
host(CRITICAL):DB Connection exception (2003, "Can't connect to MySQL server on 'localhost' ([Errno 111] Connection refused)")
host(INFO):Using pure python SQL client
host(CRITICAL):ERROR: Could not connect to MySQL database! (2003, "Can't connect to MySQL server on 'localhost' ([Errno 111] Connection refused)")
host(CRITICAL):DB Connection exception (2003, "Can't connect to MySQL server on 'localhost' ([Errno 111] Connection refused)")
host(WARNING):Too many DB failures (6), attempting to release master
host(INFO):Using pure python SQL client
host(CRITICAL):ERROR: Could not connect to MySQL database! (2003, "Can't connect to MySQL server on 'localhost' ([Errno 111] Connection refused)")
host(CRITICAL):DB Connection exception (2003, "Can't connect to MySQL server on 'localhost' ([Errno 111] Connection refused)")
host(WARNING):Too many DB failures (7), attempting to release master
host(INFO):Using pure python SQL client
Billing(INFO):Polling billing
Billing(INFO):Completed billing run for poll in 0.38s
Alerting(INFO):Checking alerts
Thread-5(CRITICAL):DB Connection exception (2013, 'Lost connection to MySQL server during query')
Thread-5(CRITICAL):DB Exception ((2013, 'Lost connection to MySQL server during query'))
performance(INFO):Counting up time spent polling
performance(CRITICAL):DB Connection exception (2013, 'Lost connection to MySQL server during query')
performance(ERROR):Unable to log performance statistics - is the database still online?
Traceback (most recent call last):
  File "/opt/librenms/LibreNMS/service.py", line 483, in log_performance_stats
    .format(self.config.node_id, self.config.name, "librenms-service", ','.join(str(g) for g in self.config.group), 1 if self.is_master else 0))
  File "/opt/librenms/LibreNMS/__init__.py", line 110, in query
    cursor.execute(query, args)
  File "/usr/local/lib/python3.6/dist-packages/pymysql/cursors.py", line 170, in execute
    result = self._query(query)
  File "/usr/local/lib/python3.6/dist-packages/pymysql/cursors.py", line 328, in _query
    conn.query(q)
  File "/usr/local/lib/python3.6/dist-packages/pymysql/connections.py", line 517, in query
    self._affected_rows = self._read_query_result(unbuffered=unbuffered)
  File "/usr/local/lib/python3.6/dist-packages/pymysql/connections.py", line 732, in _read_query_result
    result.read()
  File "/usr/local/lib/python3.6/dist-packages/pymysql/connections.py", line 1075, in read
    first_packet = self.connection._read_packet()
  File "/usr/local/lib/python3.6/dist-packages/pymysql/connections.py", line 668, in _read_packet
    "Lost connection to MySQL server during query")
pymysql.err.OperationalError: (2013, 'Lost connection to MySQL server during query')
Billing(INFO):Calculating billing
Alerting(INFO):Completed alerting run for alerts in 0.36s
Billing(INFO):Completed billing run for calculate in 0.34s
Poller_0-52(INFO):Polling device 1
Poller_0-52(INFO):Completed poller run for 1 in 5.57s
Alerting(INFO):Checking alerts
Billing(INFO):Calculating billing
Alerting(INFO):Completed alerting run for alerts in 0.33s
Billing(INFO):Completed billing run for calculate in 0.32s
```

**Comment by murrant** [comment]
Seems to reconnect for me.  Let me know if you can show otherwise.

**Comment by crazychengmm** [comment]
Aug 12 09:04:52 localhost librenms-service.py[20337]: Discovery_1-14(INFO):Device 3 is down, cannot discover, waiting 60s for retry
Aug 12 09:04:52 localhost librenms-service.py[20337]: Poller_1-3(WARNING):Polling device 3 unreachable, waiting 60s for retry
Aug 12 09:04:52 localhost librenms-service.py[20337]: Discovery_1-8(INFO):Device 2 is down, cannot discover, waiting 60s for retry
Aug 12 09:04:52 localhost librenms-service.py[20337]: Poller_1-22(WARNING):Polling device 2 unreachable, waiting 60s for retry
Aug 12 09:05:00 localhost librenms-service.py[20337]: Alerting(INFO):Checking alerts
Aug 12 09:05:00 localhost librenms-service.py[20337]: Billing(INFO):Calculating billing
Aug 12 09:05:00 localhost librenms-service.py[20337]: performance(INFO):Counting up time spent polling
Aug 12 09:05:00 localhost librenms-service.py[20337]: Billing(INFO):Calculating billing complete
Aug 12 09:05:30 localhost systemd[1]: Stopping MySQL Community Server...
Aug 12 09:05:31 localhost librenms-service.py[20337]: librenms-test1(CRITICAL):ERROR: Mysql databse has problem !(2013, 'Lost connection to MySQL server during query')
Aug 12 09:05:33 localhost systemd[1]: Stopped MySQL Community Server.
Aug 12 09:05:51 localhost librenms-service.py[20337]: librenms-test1(CRITICAL):ERROR: Database is down. (2003, "Can't connect to MySQL server on 'localhost' ([Errno 111] Connection refused)")
Aug 12 09:05:51 localhost librenms-service.py[20337]: librenms-test1(INFO):Using pure python SQL client
Aug 12 09:05:51 localhost librenms-service.py[20337]: librenms-test1(CRITICAL):ERROR: Mysql databse has problem !(2003, "Can't connect to MySQL server on 'localhost' ([Errno 111] Connection refused)")
Aug 12 09:06:00 localhost librenms-service.py[20337]: Alerting(INFO):Checking alerts
Aug 12 09:06:00 localhost librenms-service.py[20337]: Billing(INFO):Calculating billing
Aug 12 09:06:00 localhost librenms-service.py[20337]: Billing(ERROR):Billing poller script error! ('/usr/bin/env', 'php', '/home/librenms/billing-calculate.php') returned 2: b'MySQL Error: Could not connect to localhost\n'
Aug 12 09:06:00 localhost librenms-service.py[20337]: Alerting(ERROR):Alerting poller script error! ('/usr/bin/env', 'php', '/home/librenms/alerts.php') returned 2: b'MySQL Error: Could not connect to localhost\n'
Aug 12 09:06:11 localhost librenms-service.py[20337]: librenms-test1(INFO):Using pure python SQL client
Aug 12 09:06:11 localhost librenms-service.py[20337]: librenms-test1(CRITICAL):ERROR: Mysql databse has problem !(2003, "Can't connect to MySQL server on 'localhost' ([Errno 111] Connection refused)")
Aug 12 09:06:31 localhost librenms-service.py[20337]: librenms-test1(INFO):Using pure python SQL client
Aug 12 09:06:31 localhost librenms-service.py[20337]: librenms-test1(CRITICAL):ERROR: Mysql databse has problem !(2003, "Can't connect to MySQL server on 'localhost' ([Errno 111] Connection refused)")
Aug 12 09:06:51 localhost librenms-service.py[20337]: librenms-test1(INFO):Using pure python SQL client
Aug 12 09:06:51 localhost librenms-service.py[20337]: librenms-test1(CRITICAL):ERROR: Mysql databse has problem !(2003, "Can't connect to MySQL server on 'localhost' ([Errno 111] Connection refused)")
Aug 12 09:07:00 localhost librenms-service.py[20337]: Billing(INFO):Calculating billing
Aug 12 09:07:00 localhost librenms-service.py[20337]: Alerting(INFO):Checking alerts
Aug 12 09:07:00 localhost librenms-service.py[20337]: Billing(ERROR):Billing poller script error! ('/usr/bin/env', 'php', '/home/librenms/billing-calculate.php') returned 2: b'MySQL Error: Could not connect to localhost\n'
Aug 12 09:07:00 localhost librenms-service.py[20337]: Alerting(ERROR):Alerting poller script error! ('/usr/bin/env', 'php', '/home/librenms/alerts.php') returned 2: b'MySQL Error: Could not connect to localhost\n'
Aug 12 09:07:11 localhost librenms-service.py[20337]: librenms-test1(INFO):Using pure python SQL client
Aug 12 09:07:11 localhost librenms-service.py[20337]: librenms-test1(CRITICAL):ERROR: Mysql databse has problem !(2003, "Can't connect to MySQL server on 'localhost' ([Errno 111] Connection refused)")
Aug 12 09:07:31 localhost librenms-service.py[20337]: librenms-test1(INFO):Using pure python SQL client
Aug 12 09:07:31 localhost librenms-service.py[20337]: librenms-test1(CRITICAL):ERROR: Mysql databse has problem !(2003, "Can't connect to MySQL server on 'localhost' ([Errno 111] Connection refused)")
Aug 12 09:07:51 localhost librenms-service.py[20337]: librenms-test1(INFO):Using pure python SQL client
Aug 12 09:07:51 localhost librenms-service.py[20337]: librenms-test1(CRITICAL):ERROR: Mysql databse has problem !(2003, "Can't connect to MySQL server on 'localhost' ([Errno 111] Connection refused)")
Aug 12 09:08:00 localhost librenms-service.py[20337]: Alerting(INFO):Checking alerts
Aug 12 09:08:00 localhost librenms-service.py[20337]: Billing(INFO):Calculating billing
Aug 12 09:08:00 localhost librenms-service.py[20337]: Alerting(ERROR):Alerting poller script error! ('/usr/bin/env', 'php', '/home/librenms/alerts.php') returned 2: b'MySQL Error: Could not connect to localhost\n'
Aug 12 09:08:00 localhost librenms-service.py[20337]: Billing(ERROR):Billing poller script error! ('/usr/bin/env', 'php', '/home/librenms/billing-calculate.php') returned 2: b'MySQL Error: Could not connect to localhost\n'
Aug 12 09:08:11 localhost librenms-service.py[20337]: librenms-test1(INFO):Using pure python SQL client
Aug 12 09:08:11 localhost librenms-service.py[20337]: librenms-test1(CRITICAL):ERROR: Mysql databse has problem !(2003, "Can't connect to MySQL server on 'localhost' ([Errno 111] Connection refused)")
Aug 12 09:08:31 localhost librenms-service.py[20337]: librenms-test1(INFO):Using pure python SQL client
Aug 12 09:08:31 localhost librenms-service.py[20337]: librenms-test1(CRITICAL):ERROR: Mysql databse has problem !(2003, "Can't connect to MySQL server on 'localhost' ([Errno 111] Connection refused)")
Aug 12 09:08:51 localhost librenms-service.py[20337]: librenms-test1(INFO):Using pure python SQL client
Aug 12 09:08:51 localhost librenms-service.py[20337]: librenms-test1(CRITICAL):ERROR: Mysql databse has problem !(2003, "Can't connect to MySQL server on 'localhost' ([Errno 111] Connection refused)")
Aug 12 09:09:00 localhost librenms-service.py[20337]: Alerting(INFO):Checking alerts
Aug 12 09:09:00 localhost librenms-service.py[20337]: Billing(INFO):Calculating billing
Aug 12 09:09:00 localhost librenms-service.py[20337]: Alerting(ERROR):Alerting poller script error! ('/usr/bin/env', 'php', '/home/librenms/alerts.php') returned 2: b'MySQL Error: Could not connect to localhost\n'
Aug 12 09:09:00 localhost librenms-service.py[20337]: Billing(ERROR):Billing poller script error! ('/usr/bin/env', 'php', '/home/librenms/billing-calculate.php') returned 2: b'MySQL Error: Could not connect to localhost\n'
Aug 12 09:09:01 localhost CRON[22946]: (root) CMD (  [ -x /usr/lib/php/sessionclean ] && if [ ! -d /run/systemd/system ]; then /usr/lib/php/sessionclean; fi)
Aug 12 09:09:04 localhost systemd[1]: Starting Clean php session files...
Aug 12 09:09:04 localhost systemd[1]: Started Clean php session files.
Aug 12 09:09:11 localhost librenms-service.py[20337]: librenms-test1(INFO):Using pure python SQL client
Aug 12 09:09:11 localhost librenms-service.py[20337]: librenms-test1(CRITICAL):ERROR: Mysql databse has problem !(2003, "Can't connect to MySQL server on 'localhost' ([Errno 111] Connection refused)")
Aug 12 09:09:31 localhost librenms-service.py[20337]: librenms-test1(INFO):Using pure python SQL client
Aug 12 09:09:31 localhost librenms-service.py[20337]: librenms-test1(CRITICAL):ERROR: Mysql databse has problem !(2003, "Can't connect to MySQL server on 'localhost' ([Errno 111] Connection refused)")
Aug 12 09:09:51 localhost librenms-service.py[20337]: librenms-test1(INFO):Using pure python SQL client
Aug 12 09:09:51 localhost librenms-service.py[20337]: librenms-test1(CRITICAL):ERROR: Mysql databse has problem !(2003, "Can't connect to MySQL server on 'localhost' ([Errno 111] Connection refused)")
Aug 12 09:10:00 localhost librenms-service.py[20337]: Thread-6(CRITICAL):ERROR: Database is down. (2003, "Can't connect to MySQL server on 'localhost' ([Errno 111] Connection refused)")
Aug 12 09:10:00 localhost librenms-service.py[20337]: Thread-6(INFO):Using pure python SQL client
Aug 12 09:10:00 localhost librenms-service.py[20337]: Thread-6(CRITICAL):ERROR: Mysql databse has problem !(2003, "Can't connect to MySQL server on 'localhost' ([Errno 111] Connection refused)")
Aug 12 09:10:00 localhost librenms-service.py[20337]: Alerting(INFO):Checking alerts
Aug 12 09:10:00 localhost librenms-service.py[20337]: Billing(INFO):Calculating billing
Aug 12 09:10:00 localhost librenms-service.py[20337]: performance(INFO):Counting up time spent polling
Aug 12 09:10:00 localhost librenms-service.py[20337]: performance(CRITICAL):ERROR: Database is down. (2003, "Can't connect to MySQL server on 'localhost' ([Errno 111] Connection refused)")
Aug 12 09:10:00 localhost librenms-service.py[20337]: performance(INFO):Using pure python SQL client
Aug 12 09:10:00 localhost librenms-service.py[20337]: performance(CRITICAL):ERROR: Mysql databse has problem !(2003, "Can't connect to MySQL server on 'localhost' ([Errno 111] Connection refused)")
Aug 12 09:10:00 localhost librenms-service.py[20337]: Billing(ERROR):Billing poller script error! ('/usr/bin/env', 'php', '/home/librenms/billing-calculate.php') returned 2: b'MySQL Error: Could not connect to localhost\n'
Aug 12 09:10:00 localhost librenms-service.py[20337]: Alerting(ERROR):Alerting poller script error! ('/usr/bin/env', 'php', '/home/librenms/alerts.php') returned 2: b'MySQL Error: Could not connect to localhost\n'
Aug 12 09:10:11 localhost librenms-service.py[20337]: librenms-test1(INFO):Using pure python SQL client
Aug 12 09:10:11 localhost librenms-service.py[20337]: librenms-test1(CRITICAL):ERROR: Mysql databse has problem !(2003, "Can't connect to MySQL server on 'localhost' ([Errno 111] Connection refused)")
Aug 12 09:10:20 localhost librenms-service.py[20337]: Thread-6(INFO):Using pure python SQL client
Aug 12 09:10:20 localhost librenms-service.py[20337]: Thread-6(CRITICAL):ERROR: Mysql databse has problem !(2003, "Can't connect to MySQL server on 'localhost' ([Errno 111] Connection refused)")
Aug 12 09:10:20 localhost librenms-service.py[20337]: performance(INFO):Using pure python SQL client
Aug 12 09:10:20 localhost librenms-service.py[20337]: performance(CRITICAL):ERROR: Mysql databse has problem !(2003, "Can't connect to MySQL server on 'localhost' ([Errno 111] Connection refused)")
Aug 12 09:10:31 localhost librenms-service.py[20337]: librenms-test1(INFO):Using pure python SQL client
Aug 12 09:10:31 localhost librenms-service.py[20337]: librenms-test1(CRITICAL):ERROR: Mysql databse has problem !(2003, "Can't connect to MySQL server on 'localhost' ([Errno 111] Connection refused)")
Aug 12 09:10:40 localhost librenms-service.py[20337]: Thread-6(INFO):Using pure python SQL client
Aug 12 09:10:40 localhost librenms-service.py[20337]: Thread-6(CRITICAL):ERROR: Mysql databse has problem !(2003, "Can't connect to MySQL server on 'localhost' ([Errno 111] Connection refused)")
Aug 12 09:10:40 localhost librenms-service.py[20337]: performance(INFO):Using pure python SQL client
Aug 12 09:10:40 localhost librenms-service.py[20337]: performance(CRITICAL):ERROR: Mysql databse has problem !(2003, "Can't connect to MySQL server on 'localhost' ([Errno 111] Connection refused)")
Aug 12 09:10:51 localhost librenms-service.py[20337]: librenms-test1(INFO):Using pure python SQL client
Aug 12 09:10:51 localhost librenms-service.py[20337]: librenms-test1(CRITICAL):ERROR: Mysql databse has problem !(2003, "Can't connect to MySQL server on 'localhost' ([Errno 111] Connection refused)")
Aug 12 09:11:00 localhost librenms-service.py[20337]: Alerting(INFO):Checking alerts
Aug 12 09:11:00 localhost librenms-service.py[20337]: Billing(INFO):Calculating billing
Aug 12 09:11:00 localhost librenms-service.py[20337]: Thread-6(INFO):Using pure python SQL client
Aug 12 09:11:00 localhost librenms-service.py[20337]: Thread-6(CRITICAL):ERROR: Mysql databse has problem !(2003, "Can't connect to MySQL server on 'localhost' ([Errno 111] Connection refused)")
Aug 12 09:11:00 localhost librenms-service.py[20337]: performance(INFO):Using pure python SQL client
Aug 12 09:11:00 localhost librenms-service.py[20337]: performance(CRITICAL):ERROR: Mysql databse has problem !(2003, "Can't connect to MySQL server on 'localhost' ([Errno 111] Connection refused)")
Aug 12 09:11:00 localhost librenms-service.py[20337]: Billing(ERROR):Billing poller script error! ('/usr/bin/env', 'php', '/home/librenms/billing-calculate.php') returned 2: b'MySQL Error: Could not connect to localhost\n'
Aug 12 09:11:00 localhost librenms-service.py[20337]: Alerting(ERROR):Alerting poller script error! ('/usr/bin/env', 'php', '/home/librenms/alerts.php') returned 2: b'MySQL Error: Could not connect to localhost\n'
Aug 12 09:11:11 localhost librenms-service.py[20337]: librenms-test1(INFO):Using pure python SQL client
Aug 12 09:11:11 localhost librenms-service.py[20337]: librenms-test1(CRITICAL):ERROR: Mysql databse has problem !(2003, "Can't connect to MySQL server on 'localhost' ([Errno 111] Connection refused)")
Aug 12 09:11:20 localhost librenms-service.py[20337]: Thread-6(INFO):Using pure python SQL client
Aug 12 09:11:20 localhost librenms-service.py[20337]: Thread-6(CRITICAL):ERROR: Mysql databse has problem !(2003, "Can't connect to MySQL server on 'localhost' ([Errno 111] Connection refused)")
Aug 12 09:11:20 localhost librenms-service.py[20337]: performance(INFO):Using pure python SQL client
Aug 12 09:11:20 localhost librenms-service.py[20337]: performance(CRITICAL):ERROR: Mysql databse has problem !(2003, "Can't connect to MySQL server on 'localhost' ([Errno 111] Connection refused)")
Aug 12 09:11:27 localhost systemd[1]: Starting MySQL Community Server...
Aug 12 09:11:27 localhost systemd[1]: Started MySQL Community Server.
Aug 12 09:11:31 localhost librenms-service.py[20337]: librenms-test1(INFO):Using pure python SQL client
Aug 12 09:11:31 localhost librenms-service.py[20337]: Poller_1-22(INFO):Polling device 3
Aug 12 09:11:31 localhost librenms-service.py[20337]: Discovery_1-10(INFO):Discovering device 3
Aug 12 09:11:31 localhost librenms-service.py[20337]: Poller_1-19(INFO):Polling device 1
Aug 12 09:11:31 localhost librenms-service.py[20337]: Poller_1-21(INFO):Polling device 2
Aug 12 09:11:31 localhost librenms-service.py[20337]: Discovery_1-9(INFO):Discovering device 2
Aug 12 09:11:33 localhost librenms-service.py[20337]: Discovery_1-10(INFO):Device 3 is down, cannot discover, waiting 60s for retry
Aug 12 09:11:33 localhost librenms-service.py[20337]: Discovery_1-9(INFO):Device 2 is down, cannot discover, waiting 60s for retry
Aug 12 09:11:33 localhost librenms-service.py[20337]: Poller_1-22(WARNING):Polling device 3 unreachable, waiting 60s for retry
Aug 12 09:11:33 localhost librenms-service.py[20337]: Poller_1-21(WARNING):Polling device 2 unreachable, waiting 60s for retry
Aug 12 09:11:40 localhost librenms-service.py[20337]: Thread-6(INFO):Using pure python SQL client
Aug 12 09:11:40 localhost librenms-service.py[20337]: performance(INFO):Using pure python SQL client
Aug 12 09:12:00 localhost librenms-service.py[20337]: Billing(INFO):Calculating billing
Aug 12 09:12:00 localhost librenms-service.py[20337]: Alerting(INFO):Checking alerts
Aug 12 09:12:00 localhost librenms-service.py[20337]: Billing(INFO):Calculating billing complete
Aug 12 09:12:18 localhost librenms-service.py[20337]: Poller_1-19(INFO):Polling complete 1

**Comment by crazychengmm** [comment]
Hi I have tested that.It works for me.
If db restarted, i think service is better to reconnect the database.

**Comment by laf** [comment]
We're not running the latest version at work (scared to upgrade), however with 1.48 it wouldn't reconnect in the time it takes our MySQL service to restart.

**Comment by murrant** [comment]
@laf it was reworked since then.  You are really behind now :D

**Pull request closed without merging**